### PR TITLE
[wip] Travis -> Cirrus

### PR DESCRIPTION
**Pull request by q3k**
We replace Travis with Cirrus as Travis keeps giving us trouble, and Cirrus is generally a better alternative (notably, supports running in Docker containers for Linux, and supports Windows in addition to macOS).

### CODE REVIEW

File: .cirrus/Dockerfile.ubuntu14.04 [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+RUN set -e -x ;\
+    cd /tmp ;\
+    git clone git://github.com/steveicarus/iverilog.git ;\
+    cd iverilog ;\
+    autoconf ;\
+    ./configure ;\
+    make -j $(nproc) ;\
+    make install ;\
+    cd .. ;\
+    rm -rf iverilog
```
Comment (whitequark) [inline .cirrus/Dockerfile.ubuntu14.04]: Is... is this really how you are supposed to use Docker? Is this *really* better than Travis?..

Comment (q3k) [reply .cirrus/Dockerfile.ubuntu14.04]: Yes.

**Comment by cliffordwolf** [comment]
What's the current status of this? I'm very much looking forward to having a reliable CI system sooner than later..

**Comment by nakengelhardt** [comment]
This looks abandoned. I'll close this PR and we can open a new one if anyone ever takes it up again.

**Pull request closed without merging**